helpers/httpApi.js

```diff
@@ -140,6 +140,60 @@ var middleware = {
 	attachResponseHeaders: function (getHeaders, req, res, next) {
 		res.set(getHeaders());
 		return next();
+	},
+
+	/**
+	 * Lookup cache, and reply with cached response if it's a hit.
+	 * If it's a miss, forward the request but cache the response if it's a success.
+	 * @param {Object} req
+	 * @param {Object} res
+	 * @param {Function} next
+	 */
+	useCache: function (logger, cache, req, res, next) {
+		if (cache.isConnected()) {
+			logger.info('Req parameters', req.url, JSON.stringify(req.params));
+			var key = req.url + JSON.stringify(req.params);
+
+			cache.getJsonForKey(key, function (err, cachedValue) {
+				//there was an error or value doesn't exist for key
+				if (err || !cachedValue) {
+					// Monkey patching res.json function only if we expect to cache response
+					var expressSendJson = res.json;
+					res.json = function (response) {
+						if (response.success) {
+							cache.setJsonForKey(key,response);
+							expressSendJson.call(res, response);
+						} else {
+							logger.info('serving response for url: ', req.url, ' from cache');
+							expressSendJson.call(res, response);
+						}
+					};
+					next();
+				} else {
+					res.json(cachedValue);
+				}
+			});
+		} else {
+			next();
+		}
+	},
+	flushCache: function (logger, cache, req, res, next) {
```

Comment:
So where is `flushCache` action invoked? Where are we invalidating data in other words